tools/cleanup/cleanup-functions.sh

```diff
@@ -48,21 +48,51 @@ function delete_old_images_from_gcr() {
   done
 }
 
-# Delete old images in the GCP projects defined in the yaml file provided.
-# Parameters: $1 - yaml file path defining projects that will be cleaned up
-#             $2 - regex pattern for parsing the project names
-#             $3 - days to keep images
+# Delete old images in the given GCP projects
+# Parameters: $1 - array of projects names
+#             $2 - days to keep images
 function delete_old_gcr_images() {
-  [[ -z $1 ]] && abort "missing resource yaml path"
-  [[ -z $2 ]] && abort "missing regex pattern for project name"
-  [[ -z $3 ]] && abort "missing days to keep images"
-
-  local target_projects # delared here as local + assignment in one line always return 0 exit code
-  target_projects="$(grep -Eio "$2" "$1")"
-  [[ $? -eq 0 ]] || abort "no project found in $1"
+  [[ -z $1 ]] && abort "missing project names"
+  [[ -z $2 ]] && abort "missing days to keep images"
 
-  for project in ${target_projects}; do
+  for project in $1; do
     echo "Start deleting images from ${project}"
-    delete_old_images_from_gcr "gcr.io/${project}" $3
+    delete_old_images_from_gcr "gcr.io/${project}" $2
+  done
+}
+
+# Delete old clusters in the given GCP projects
+# Parameters: $1 - array of projects names
+#             $2 - hours to keep images
+function delete_old_test_clusters() {
+  [[ -z $1 ]] && abort "missing project names"
+  [[ -z $2 ]] && abort "missing hours to keep clusters"
+
+  for project in $1; do
+    echo "Start deleting clusters from ${project}"
+
+    is_protected_project $project && \
+      abort "Target project set to $project, which is forbidden"
+
+    local current_time=$(date +%s)
+    local target_time=$(date -d "`date -d @${current_time}`-$2hours" +%s)
+    # Fail if the difference of current time and target time is not 3600 times hours to keep
```

Comment:
Why this check? Did I miss something in the tests?

Reply:
This might not be necessary, just want to ensure that the second `date` command always does what it supposed to be: get a timestamp 2 hours before `${current_time}. 
Check it here instead of in unit testing as I don't think this should fail PR.

Reply:
Sorry, I'm failing to see why you check for dry-run here. Your tests only use --dry-run when checking for wrong/missing values, so they never reach this point.

Reply:
@chaodaiG I don't think we clarified this point?

Reply:
sorry I thought we did. Now I understand the concern better, will get back to address this when I got a chance

Reply:
@adrcunha , I just double checked, the last test does reach this point, and it's the reason why `DRY_RUN` is checked here

Reply:
Please clearly document the reason here. "but not blocking PR" is obscure. I suggest something like "Don't check on dry runs, as dry run is used for unit testing".

Reply:
done